src/modules/claim/services/claim-issuance.service.ts

```diff
@@ -83,16 +86,70 @@ export class ClaimIssuanceService {
       claimType: dto.claimType,
     });
 
-    await this.roleService.verifyEnrolmentPrecondition({
-      claimType: dto.claimType,
-      userDID: dto.requester,
-    });
+    const enrolmentPreconditions =
+      await this.roleService.fetchEnrolmentPreconditions({
+        claimType: dto.claimType,
+      });
+    if (
+      enrolmentPreconditions?.length > 0 &&
+      enrolmentPreconditions.find((precond) => precond.type === 'role')
```

Comment:
This line is interesting. 
- At a minimum, it should be replaced with the enum rather than the string
- IMO, we need to add an else statement that throws error if the precond type is not 'role'. Otherwise, if another type is added and someone runs ssi-hub that doesn't recognize it, enrolmentPrecondition check could be bypassed. IMO, it is better to err on the side of safety here.
- Given the above point, I think we should use `every` and not `find`.
```suggestion
      enrolmentPreconditions.every((precond) => precond.type === PreconditionType.Role)
```

Reply:
used enum, made .every, made error message to throw if .every was not met and enrolmentConditions.length